libs/ledger-live-common/src/device/use-cases/appDataBackup/types.ts

```diff
@@ -0,0 +1,59 @@
+import { type AppStorageInfo, appStorageInfoGuard } from "@ledgerhq/device-core";
+import { DeviceModelId } from "@ledgerhq/devices";
+import { createCustomErrorClass } from "@ledgerhq/errors";
+
+export interface StorageProvider<VType> {
+  getItem(key: string): Promise<VType | null>;
+  setItem(key: string, value: VType): Promise<void>;
+  removeItem(key: string): Promise<void>;
+}
+
+export type AppName = string;
+export type AppStorageKey =
+  `${Exclude<DeviceModelId, DeviceModelId.blue | DeviceModelId.nanoS>}-${AppName}`;
```

Comment:
[Ask] is it really necessary to exclude the blue & nanoS here?

Reply:
I've forgotten why they are excluded, but if they are support to backup data, I think I need to add them back. I will check.

Reply:
yes but even if they are not supported I'm not sure it's necessary to narrow the type of the storage key 🤷‍♂️